### ISSUE TITLE
Incompatibility with custom authorization strategies & GraphQL::Analysis::AST / GraphQL::Pro::OperationStore

### ISSUE THREAD

**Issue by plasticine** [rmosolgo/graphql-ruby#2935]
Hey there 👋 Thanks for GraphQL / GraphQL::Pro — they’re awesome!

We’ve recently been looking to upgrade a few of our graphql schemas to use `GraphQL::Execution::Interpreter`, `GraphQL::Analysis::AST`, and `GraphQL::Pro::OperationStore` for our main schema, but we’re seeing some really confusing and inconsistent results with using custom authentication strategies.

**Versions**

- `graphql` version: `1.10.9`
- `graphql-pro` version: `1.13.4`
- `graphql-batch` version: `0.4.2`
- `rails` version: `5.2.4.2`

**GraphQL schema**

```ruby
class Graph::Schema < GraphQL::Schema
  query Graph::Types::QueryType
  mutation Graph::Types::MutationType

  use GraphQL::Execution::Interpreter
  use GraphQL::Batch
  use GraphQL::Tracing::DataDogTracing, service: "api-graphql"
  use GraphQL::Analysis::AST

  authorization Graph::AuthStrategy, fallback: { authorize: :logged_in }

  use GraphQL::Pro::OperationStore

  # Snip!
end
```

**Steps to reproduce**

Configure a schema with a custom auth strategy and `GraphQL::Execution::Interpreter` & `GraphQL::Analysis::AST`. The custom strategy will never be used.

**Expected behavior**

Custom authorization strategy should be invoked.

**Actual behavior**

Currently the schema above is yielding;

```
undefined method `new' for GraphQL::Pro::Access::Analyzer:Module
```

...when executed (`Graph::AuthStrategy` is pretty straightforward and does nothing particularly fancy).

What’s even more confusing is that if `use GraphQL::Pro::OperationStore` is removed from the schema then the error goes away, but then `Graph::AuthStrategy` is never invoked, and `Graph::Schema.query_analyzers` is empty.

I’m not sure if I’m PEBKAC-ing the schema configuration? — is `authorization` support changed in some way I’m not accounting for with `GraphQL::Analysis::AST` here that I’m missing?

<details>
<summary>Click to view exception backtrace</summary>
```
NoMethodError - undefined method `new' for GraphQL::Pro::Access::Analyzer:Module:
  app/controllers/graph_controller.rb:25:in `block in execute!'
  app/controllers/graph_controller.rb:24:in `execute!'
  app/controllers/graph_controller.rb:8:in `graphiql'
  lib/logging/logger.rb:27:in `tagged'
  lib/middleware/clear_log_context.rb:10:in `call'
```
</details>

**Comment by rmosolgo** [comment]
Hey, thanks for the really detailed writeup. 

> is `authorization` support changed in some way

Yes, that's basically it. That system is mostly replaced by a new system that's integrated into class-based schema definitions: https://graphql-ruby.org/authorization/overview.html#graphql-rubys-authorization-framework

Sorry there wasn't clearer error messaging there 😖 Most folks were using Pundit or CanCan integrations, and those have dedicated replacements in graphql-pro. If you want to share (or email me) your auth strategy, I'd be happy to make a suggestion about how it could be migrated to the class-based authorization system.

**Comment by rmosolgo** [comment]
Happy to provide more help migrating here if you're interested. Just leave a comment or shoot me an email!

**Comment by yutamgit** [comment]
Hi again @rmosolgo 👋 

I've been encountering the `undefined method 'new' for GraphQL::Pro::Access::Analyzer:Module` error, too. I removed the `authorization(:pundit, ...)` line we used to have and started authorizing at the field/type levels. 

From the thread above, it seems like using `authorization` at the schema-level is not supported anymore. It still is confusing that `authorization` is a valid definition. Just curious if `AuthorizationDefinition` will be removed in future pro versions to avoid the confusion. 

Thank you again for the support!
